zio-json/shared/src/main/scala-2.x/zio/json/macros.scala

```diff
@@ -508,7 +512,9 @@ object DeriveJsonEncoder {
           config.explicitNulls || ctx.annotations.exists(_.isInstanceOf[jsonExplicitNull])
 
         val explicitEmptyCollections: Boolean =
-          config.explicitEmptyCollections || ctx.annotations.exists(_.isInstanceOf[jsonExplicitEmptyCollection])
+          ctx.annotations.collectFirst { case jsonExplicitEmptyCollection(enabled) =>
+            enabled
+          }.getOrElse(config.explicitEmptyCollections)
```

Comment:
Annotations first then config. This way implicit config cannot change the codec unintentionally. The JsonCodecConfiguration is only for when no annotations are present.

Reply:
I am not sure about this. You can use different configs in different places. But there is only one annotation

Reply:
I cannot find any documentation saying anything about priorities of the JsonCodecConfiguration. I would say it is just another default but annotations take precedence. 
I just checked the other configs and all are using annotations first and default to the JsonCodecConfiguration if no annotations are found. The explicitNulls config looks like to be the exception and should perhaps be changed.